src/main/scala/scalismo/faces/sampling/face/proposals/SphericalHarmonicsLightProposals.scala

```diff
@@ -264,4 +269,133 @@ object SphericalHarmonicsLightProposals {
 
     override def toString = s"SHLightSolverProposal($sphericalHarmonicsOptimizer)"
   }
+
+  /** This proposal performs RANSAC to try to estimate illumination excluding outliers/occlusions
+    * The face model is used as appearance prior
+    *
+    * The procedure is explained in:
+    * Occlusion-aware 3D Morphable Face Models,
+    * Bernhard Egger, Andreas Schneider, Clemens Blumer, Andreas Morel-Forster, Sandro Schönborn, Thomas Vetter
+    * IN: British Machine Vision Conference (BMVC), September 2016
+    * https://dx.doi.org/10.5244/C.30.64
+    * */
+  case class  RobustSHLightSolverProposal(modelRenderer: ParametricImageRenderer[RGBA] with ParametricModel,
+                                          shOpt: SphericalHarmonicsOptimizer,
+                                          pixelEvaluator: PairEvaluator[RGB],
+                                          target: PixelImage[RGBA],
+                                          nSamples: Int = 30,
+                                          sigmaThreshold: Double = 2,
+                                          percentage: Double = 0.4,
+                                          iterations: Int = 500,
+                                          updateLight: Boolean = true,
+                                          maxVarianceForMask: Double = 0.5,
+                                          nSamplesIllumination: Int = 1000)(implicit rnd: Random)
+    extends ProposalGenerator[(RenderParameter, PixelImage[Int])] with TransitionProbability[(RenderParameter, PixelImage[Int])] {
+
+
+
+    // the normalizer for the given pixel evaluator
+    val normalizer = pixelEvaluator.logValue(RGB.White, RGB.White)
+    // the corresponding threshold depending on sigmaThreshold
```

Comment:
(comment without additional information as it is obvious in the code: `val threshold = .... sigmaThreshold...` )